mergin/test/test_client.py

```diff
@@ -805,3 +805,71 @@ def test_logging(mc):
 def test_server_compatibility(mc):
     """Test server compatibility."""
     assert mc.is_server_compatible()
+
+
+def create_versioned_project(mc, project_name, project_dir, updated_file, remove=True):
+    project = API_USER + '/' + project_name
+    cleanup(mc, project, [project_dir])
+
+    # create remote project
+    shutil.copytree(TEST_DATA_DIR, project_dir)
+    mc.create_project_and_push(project_name, project_dir)
+
+    mp = MerginProject(project_dir)
+
+    # create versions 2-5
+    changes = ("inserted_1_A.gpkg", "inserted_1_A_mod.gpkg", "inserted_1_B.gpkg",)
+    for change in changes:
+        shutil.copy(mp.fpath(change), mp.fpath(updated_file))
+        mc.push_project(project_dir)
+
+    if remove:
+        os.remove(os.path.join(project_dir, updated_file))
+        mc.push_project(project_dir)
+
+    return mp
+
+
+def test_download_file(mc):
+    """Test downloading single file at specified versions."""
+    test_project = 'test_download_file'
+    project = API_USER + '/' + test_project
+    project_dir = os.path.join(TMP_DIR, test_project)
+    download_dir = os.path.join(project_dir, "versions")  # project for downloading files at various versions
+    f_updated = "base.gpkg"
+
+    mp = create_versioned_project(mc, test_project, project_dir, f_updated)
+
+    project_info = mc.project_info(project)
+    assert project_info["version"] == "v5"
+
+    # Download the base file at versions 2-4 and check their checksums
+    f_versioned = os.path.join(download_dir, f_updated)
+    for ver in range(2, 5):
+        mc.download_file(project, f_updated, f_versioned, version=f"v{ver}")
+        f_downloaded_checksum = generate_checksum(f_versioned)
```

Comment:
let's not test checksums - we will be removing them on both server and clients (they are slowing things down and not that useful)